File: app/src/main/java/com/absinthe/kage/device/IDeviceObserver.java
package com.absinthe.kage.device;

import com.absinthe.kage.device.model.DeviceInfo;

public interface IDeviceObserver {

    void onFindDevice(DeviceInfo deviceInfo);
    void onLostDevice(DeviceInfo deviceInfo);
    void onDeviceConnected(DeviceInfo deviceInfo);
    void onDeviceDisConnect(DeviceInfo deviceInfo);
    void onDeviceConnectFailed(DeviceInfo deviceInfo, int errorCode, String errorMessage);
    void onDeviceInfoChanged(DeviceInfo deviceInfo);
    void onDeviceNotice(DeviceInfo deviceInfo);
    void onDeviceConnecting(DeviceInfo deviceInfo);

}

File: app/src/main/kotlin/com/absinthe/kage/device/IDeviceObserver.kt
package com.absinthe.kage.device

import com.absinthe.kage.device.model.DeviceInfo

interface IDeviceObserver {
    fun onFindDevice(deviceInfo: DeviceInfo)
    fun onLostDevice(deviceInfo: DeviceInfo)
    fun onDeviceConnected(deviceInfo: DeviceInfo)
    fun onDeviceDisConnect(deviceInfo: DeviceInfo)
    fun onDeviceConnectFailed(deviceInfo: DeviceInfo, errorCode: Int, errorMessage: String?)
    fun onDeviceInfoChanged(deviceInfo: DeviceInfo)
    fun onDeviceNotice(deviceInfo: DeviceInfo)
    fun onDeviceConnecting(deviceInfo: DeviceInfo)
}